src/mscorlib/src/System/Runtime/InteropServices/NativeLibrary.cs

```diff
@@ -0,0 +1,346 @@
+        {
+            Debug.Assert(((RuntimeType)typeof(TDelegate)).IsDelegate());
+
+            IntPtr symbolAddress = GetProcAddress(_hInstance, symbolName);
+            return (symbolAddress != IntPtr.Zero)
+                ? Marshal.GetDelegateForFunctionPointer<TDelegate>(symbolAddress)
+                : null;
+        }
+
+        // Looks up a symbol by ordinal and if it's found returns a delegate for it.
+        // Returns null if the symbol is not found.
+        private TDelegate CreateDelegateForSymbolByOrdinal<TDelegate>(ushort ordinal) where TDelegate : class
+        {
+            Debug.Assert(((RuntimeType)typeof(TDelegate)).IsDelegate());
+
+            IntPtr symbolAddress = GetProcAddress(_hInstance, (IntPtr)ordinal);
+            return (symbolAddress != IntPtr.Zero)
+                ? Marshal.GetDelegateForFunctionPointer<TDelegate>(symbolAddress)
+                : null;
+        }
+
+        /// <summary>
+        /// Equivalent to <see cref="TryGetDelegate{TDelegate}(string, bool, out TDelegate)"/> where <em>exactSpelling</em> is <see langword="false"/>.
+        /// </summary>
+        public bool TryGetDelegate<TDelegate>(string name, out TDelegate result) where TDelegate : class
+            => TryGetDelegate<TDelegate>(name, exactSpelling: false, out result);
+
+        /// <summary>
+        /// Attempts to locate a named function in the module's export table and return a managed delegate to that function.
+        /// </summary>
+        /// <typeparam name="TDelegate">The delegate type that will be created. This type must subclass <see cref="MulticastDelegate"/>.
+        /// This type <em>should</em> be (but is not required to be) annotated with <see cref="UnmanagedFunctionPointerAttribute"/>.
+        /// Generic delegate types (e.g., <see cref="Func{T, TResult}"/>) are not supported.</typeparam>
+        /// <param name="name">The name of the function to look up.</param>
+        /// <param name="exactSpelling"><see langword="false"/> (default) if the runtime should automatically perform name decoration when
+        /// locating the exported function; <see langword="true"/> if the runtime should use the provided name as-is with no additional
+        /// decoration. See also <see cref="DllImportAttribute.ExactSpelling"/>.</param>
+        /// <param name="result">A <typeparamref name="TDelegate"/> that represents the exported function,
+        /// or <see langword="null"/> if the function cannot be located in the export table.</param>
+        /// <returns><see langword="true"/> if the function was found, otherwise <see langword="false"/>.</returns>
+        /// <remarks>
+        /// Functions can also be queried by ordinal export by using a "#" prefix (e.g., "#1") for <paramref name="name"/>.
+        /// </remarks>
+        public bool TryGetDelegate<TDelegate>(string name, bool exactSpelling, out TDelegate result) where TDelegate : class
```

Comment:
Do you try to add support for SetLastError?  also others fields in DllImportAttribute

Reply:
The existing marshaling code handles that automatically. We get it for free.

Reply:
For reference, this comes from the existing `Marshal.GetDelegateForFunctionPointer` behavior. It respects any `[UnmanagedFunctionPointer]` attribute specified on the delegate type, including the _SetLastError_ field if specified. Since we're really just a glorified wrapper around `Marshal.GetDelegateForFunctionPointer` we get it for free.

Reply:
This need to add a doc(or sample code) to show customer how to do it correctly.